controllers/podchaos/containerkill/types.go

```diff
@@ -96,6 +98,10 @@ func (r *endpoint) Apply(ctx context.Context, req ctrl.Request, obj v1alpha1.Inn
 		return err
 	}
 
+	if allHaveContainer == false {
+		return fmt.Errorf("at least one pod contains no designated container")
+	}
+
 	podchaos.Status.Experiment.PodRecords = make([]v1alpha1.PodStatus, 0, len(pods))
 	for _, pod := range pods {
```

Comment:
I have a doubt: if some pods have a container and some pods do not have a container in the selected pods, at this time, if the above returns directly, the corresponding pod_records will not be recorded. Is this reasonable? Or should you record pod_records and then "return fmt.Errorf("at least one pod contains no designated container")"?

Reply:
Yes, I think we should record this pod_records.